test/translate_c.zig

```diff
@@ -59,6 +59,20 @@ pub fn addCases(cases: *tests.TranslateCContext) void {
         );
     }
 
+    cases.add("predefined expressions",
+        \\void foo(void) {
+        \\    __func__;
+        \\    __FUNCTION__;
+        \\    __PRETTY_FUNCTION__;
+        \\}
+    ,
+        \\pub fn foo() void {
+        \\    c"foo";
```

Comment:
shouldn't these be `_ = c"foo";` ? otherwise I get these compile errors:

```
/home/andy/dev/zig/build/test.zig:2:5: error: expression value is ignored
    c"foo";
    ^
/home/andy/dev/zig/build/test.zig:3:5: error: expression value is ignored
    c"foo";
    ^
/home/andy/dev/zig/build/test.zig:4:5: error: expression value is ignored
    c"void foo(void)";
    ^
```

Reply:
They should, but as I mentioned in IRC, the ignore value system doesn't take care of a lot of things. Any string literal will generate code where it isn't ignored

Reply:
Let me see if I can fiddle with it and make it work, otherwise I'll just merge this as is.

Reply:
@andrewrk I took the time to implement value ignoring the way I think makes most sense. Idk if I cover everything but it's a start.